src/pages/en/guides/content-collections.mdx

```diff
@@ -396,6 +396,127 @@ const { Content } = await entry.render();
 <Content />
 ```
 
+## Migrating from File-Based Routing 
+If your existing site uses [file-based routing](/en/core-concepts/astro-pages/#markdownmdx-pages) with Markdown files in the `src/pages` folder, you may want to migrate to content collections. This gives you the benefit of type-safe frontmatter, but requires you to set up routing yourself. This guide uses the result of the [Build a Blog tutorial](/en/tutorial/0-introduction/) as an example starting point.
+
+1. [Upgrade](/en/guides/upgrade-to/v2/) to Astro v2.0 or later, and upgrade all integrations to their latest versions. 
+
+2. [Set up TypeScript](/en/guides/content-collections/#setting-up-typescript) for content collections.
+3. Move Markdown and MDX pages from `src/pages` to subdirectories of `src/content`. Each subdirectory represents a content type. For example, to migrate the [blog posts in the tutorial](/en/tutorial/2-pages/2/), move the contents of `src/pages/posts` to `src/content/posts`. 
+4. Create a `src/content/config.ts` file and [define a schema](/en/guides/content-collections/#defining-a-collection-schema) for each content type. For the blog, we only have one content type, `posts`:
+
+```ts title="src/content/config.ts"
+// Import utilities from `astro:content`
+import { z, defineCollection } from "astro:content";
+// Define a schema for each collection you'd like to validate.
+const postsCollection = defineCollection({
+    schema: z.object({
+      title: z.string(),
+      pubDate: z.date(),
+      description: z.string(),
+      author: z.string(),
+      image: z.object({
+        url: z.string(),
+        alt: z.string()
+      }),
+      tags: z.array(z.string())
+    })
+ });
+// Export a single `collections` object to register your collection(s)
+export const collections = {
+  posts: postsCollection,
+};
+```
+
+:::tip
+If your editor doesn't recognize `astro:content`, make sure you're on the latest version of Astro and try restarting the dev server. 
+:::
+
+5. To [generate routes](/en/guides/content-collections/#generating-routes-from-content), create a `.astro` page with a [dynamic route](/en/core-concepts/routing/#dynamic-routes) in `src/pages/posts` to [query and render your content](#querying-collections). Our tutorial blog uses the static output mode, so we'll use `getStaticPaths()` to create a route for each post:
```

Comment:
```suggestion

:::note
Inside a collection, Markdown and MDX files no longer automatically create pages using Astro's [file-based routing inside `src/pages/`](/en/guides/markdown-content/#file-based-routing). You will need to [generate routes from your collections](/en/guides/content-collections/#generating-routes-from-content) yourself.
:::

5. Create a `src/pages/posts/[...slug].astro`. This page will use [dynamic routing](/en/core-concepts/routing/#dynamic-routes) to generate a page for each collection entry. 

This page will also need to [query your collection](#querying-collections) to fetch page slugs and make the page content available to each route.

Render your post `<Content />` within the layout for your Markdown or MDX pages, since they can no longer rely on the `layout` frontmatter property to specify their own layout.
```
I'm not sold on this edit, but the original felt way too dense for one step. We felt very much in "draw the rest of the owl" territory here!  :sweat_smile: 

So, I tried to emphasize routing as a special thing, and make the other points as distinct follow-ups.